mysql-test/suite/component_masking_functions/t/rpl_dictionaries_flush_interval.test

```diff
@@ -0,0 +1,138 @@
+--source include/assert.inc
+
+
+--source include/rpl_sync.inc
+--source include/rpl_connection_slave.inc
+--let $assert_cond = [SELECT COUNT(*) FROM mysql.masking_dictionaries] = 2
+--let $assert_text = The number of dictionary terms on replica after synchronization must be 2
+--source include/assert.inc
+
+--let $assert_cond = gen_dictionary("dict1") IS NULL
+--let $assert_text = The call to gen_dictionary("dict1") on replica after synchronization but without reloading cache must return NULL
+--source include/assert.inc
+
+--let $assert_cond = gen_dictionary("dict2") IS NULL
+--let $assert_text = The call to gen_dictionary("dict1") on replica after synchronization but without reloading cache must return NULL
+--source include/assert.inc
+
+SELECT masking_dictionaries_flush();
+
+--let $assert_cond = gen_dictionary("dict1") = "term1"
+--let $assert_text = The call to gen_dictionary("dict1") on replica after synchronization and reloading cache must return "term1"
+--source include/assert.inc
+
+--let $assert_cond = gen_dictionary("dict2") = "term2"
+--let $assert_text = The call to gen_dictionary("dict1") on replica after synchronization and reloading cache must return "term2"
+--source include/assert.inc
+
+
+--source include/rpl_connection_master.inc
+INSERT INTO mysql.masking_dictionaries VALUES ('dict3', 'term3');
+
+--let $assert_cond = gen_dictionary("dict3") IS NULL
+--let $assert_text = The call to gen_dictionary("dict3") on source with suspended flusher thread must return NULL
+--source include/assert.inc
+
+SET debug_sync = 'now SIGNAL masking_functions_before_cache_reload_signal WAIT_FOR masking_functions_after_cache_reload_signal';
+
+--let $assert_cond = gen_dictionary("dict3") = "term3"
+--let $assert_text = The call to gen_dictionary("dict3") on source after resuming flusher thread must return "term3"
+--source include/assert.inc
+
+
+--source include/rpl_sync.inc
+--source include/rpl_connection_slave.inc
+--let $assert_cond = [SELECT COUNT(*) FROM mysql.masking_dictionaries] = 3
+--let $assert_text = The number of dictionary terms on replica after the second synchronization must be 3
+--source include/assert.inc
+
+--let $assert_cond = gen_dictionary("dict3") IS NULL
+--let $assert_text = The call to gen_dictionary("dict3") on replica with suspended flusher thread must return NULL
+--source include/assert.inc
+
+SET debug_sync = 'now SIGNAL masking_functions_before_cache_reload_signal WAIT_FOR masking_functions_after_cache_reload_signal';
+
+--let $assert_cond = gen_dictionary("dict3") = "term3"
+--let $assert_text = The call to gen_dictionary("dict3") on replica after resuming flusher thread must return "term3"
+--source include/assert.inc
+
+#
+# Cleanup
+--source include/rpl_connection_slave.inc
+REVOKE MASKING_DICTIONARIES_ADMIN ON *.* FROM root@localhost;
+SET GLOBAL debug ='-d,enable_masking_functions_flush_thread_sync';
```

Comment:
Ditto here. I think it is a good style to reset debug_sync facility at the end of the test as well for both master and slave (though we are not likely to see any problems for this specific test due to such resetting missing).

Reply:
Done